modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementFailureTest.java

```diff
@@ -167,4 +167,91 @@ public void testDeleteNotExistingBrandingPreference() {
         Response response = getResponseOfDelete(BRANDING_PREFERENCE_API_BASE_PATH);
         validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "BPM-60002");
     }
+
+    @Test
+    public void testAddCustomTextPreferenceWithEmptyJsonPreference() throws IOException {
+
+        String body = readResource("add-empty-custom-text-preference.json");
+        Response response = getResponseOfPost(CUSTOM_TEXT_API_BASE_PATH, body);
+        validateErrorResponse(response, HttpStatus.SC_BAD_REQUEST, "BPM-60005");
+    }
+
+    @Test
+    public void testAddCustomTextPreferenceConflict() throws IOException {
+
+        String body = readResource("add-custom-text.json");
+        // Add Custom Text Preference.
+        Response response = getResponseOfPost(CUSTOM_TEXT_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+
+        // Add conflicting Custom Text Preference.
+        response = getResponseOfPost(CUSTOM_TEXT_API_BASE_PATH, body);
+        validateErrorResponse(response, HttpStatus.SC_CONFLICT, "BPM-60007");
+
+        // Delete Custom Text Preference.
+        response = getResponseOfDelete(CUSTOM_TEXT_API_BASE_PATH + QUERY_PARAM_SEPARATOR +
+                String.format(SCREEN_QUERY_PARAM, LOGIN_SCREEN));
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+    }
+
+    @Test
+    public void testGetNotExistingCustomTextPreference() {
+
+        Response response = getResponseOfGet(CUSTOM_TEXT_API_BASE_PATH + QUERY_PARAM_SEPARATOR +
+                String.format(SCREEN_QUERY_PARAM, LOGIN_SCREEN));
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "BPM-60006");
+    }
+
+    @Test
+    public void testUpdateNotExistingCustomTextPreference() throws IOException {
+
+        String body = readResource("update-custom-text.json");
+        Response response = getResponseOfPut(CUSTOM_TEXT_API_BASE_PATH, body);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "BPM-60006");
+    }
+
+    @Test
+    public void testUpdateCustomTextPreferenceWithEmptyJsonPreference() throws IOException {
```

Comment:
Can combine this test and `testAddCustomTextPreferenceConflict` with a data provider